### PR TITLE
added the previous PR changes to new branch

### PR DESCRIPTION
**Pull request by Gajesh2007**

### CODE REVIEW

**Review by Sidu28 — CHANGES_REQUESTED** [review]
This seems fine but it looks like you've left out some of the changes present in the previous PR: https://github.com/Layr-Labs/eigenlayer-contracts/pull/274/files

**Comment by Gajesh2007** [comment]
which one are u referring to?

**Comment by Sidu28** [comment]
> which one are u referring to?

you made edits to testDeployEigenPodTooSoon and testWithdrawBeforeRestaking in the previous PR, those changes aren't showing here as far as I can tell

File: src/test/EigenPod.t.sol [hunk 1]
```diff
@@ -265,6 +265,13 @@ contract EigenPodTests is ProofParsing, EigenPodPausingConstants {
         cheats.stopPrank();
     }
 
+    function testStakingWithInvalidAmount () public {
```
Comment (Sidu28) [inline src/test/EigenPod.t.sol:268]: can you move this test to EigenPodUnit.t.sol and then do the test with pod.stake instead of podManager.stake

**Review by Sidu28 — CHANGES_REQUESTED** [review]
look at my comment above ^^ otherwise lgtm

**Review by Sidu28 — APPROVED** [review]

**Review by Sidu28 — CHANGES_REQUESTED** [review]
CI failing @Gajesh2007

**Pull request merged**